RiotSwiftUI/Modules/UserSessions/UserOtherSessions/View/UserOtherSessionsToolbar.swift

```diff
@@ -0,0 +1,94 @@
+//
+// Copyright 2022 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import SwiftUI
+
+struct UserOtherSessionsToolbar: ToolbarContent {
+    @Environment(\.theme) private var theme
+    
+    @Binding var isEditModeEnabled: Bool
+    @Binding var filter: UserOtherSessionsFilter
+    var allItemsSelected: Bool
+    let onToggleSelection: () -> Void
+    
+    var body: some ToolbarContent {
+        navigationBarLeading()
+        navigationBarTrailing()
+    }
+    
+    private func navigationBarLeading() -> some ToolbarContent {
+        ToolbarItemGroup(placement: .navigationBarLeading) {
+            if isEditModeEnabled {
+                Button(allItemsSelected ? VectorL10n.deselectAll : VectorL10n.selectAll, action: {
+                    onToggleSelection()
+                })
+            }
+        }
+    }
+    
+    private func navigationBarTrailing() -> some ToolbarContent {
+        ToolbarItemGroup(placement: .navigationBarTrailing) {
+            if isEditModeEnabled {
+                cancelButton()
+            } else {
+                filterMenuButton()
+                    .offset(x: 12)
+                kebabMenu()
+            }
+        }
+    }
+    
+    private func cancelButton() -> some View {
+        Button(VectorL10n.cancel) {
+            isEditModeEnabled = false
+        }
+        .font(theme.fonts.bodySB)
+        .foregroundColor(theme.colors.accent)
+    }
+    
+    private func filterMenuButton() -> some View {
+        Button { } label: {
+            Menu {
+                Picker("", selection: $filter) {
+                    ForEach(UserOtherSessionsFilter.allCases) { filter in
+                        Text(filter.menuLocalizedName).tag(filter)
+                    }
+                }
+                .labelsHidden()
+            } label: {
+                Image(filter == .all ? Asset.Images.userOtherSessionsFilter.name : Asset.Images.userOtherSessionsFilterSelected.name)
+            }
+            .accessibilityLabel(VectorL10n.userOtherSessionFilter)
+        }
+    }
+    
+    private func kebabMenu() -> some View {
```

Comment:
Not sure about using the `kebab` in the name of this view / function. Nevermind the fact that `kebab` mostly denotes vertical dots (and `meatballs` for horizontal), it might be better to call this by its function, e.g. `optionsMenu`, or `moreMenu`, similar to `filterMenuButton`.

Reply:
ACs were using "kebab" so I stuck to it, but I will change it `optionsMenu` 👍